frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.tsx

```diff
@@ -0,0 +1,48 @@
+import _ from "underscore";
+import { isNotNull } from "metabase/core/utils/array";
+import { getColorsForValues } from "metabase/lib/colors/charts";
+import { formatStaticValue } from "metabase/static-viz/lib/format-static-value";
+import { ColorPalette } from "metabase/lib/colors/types";
+import {
+  ChartSettings,
+  Series,
+  SeriesWithBreakoutValues,
+  SeriesWithoutBreakoutValues,
+} from "../../XYChart/types";
+
+export function getSeriesWithColors(
+  multipleSeries: (SeriesWithoutBreakoutValues | SeriesWithBreakoutValues)[],
+  palette: ColorPalette,
+  settings: ChartSettings,
+): Series[] {
+  const keys = multipleSeries
+    .map(series => {
+      if (hasBreakoutValues(series)) {
+        return formatStaticValue(series.name, {
+          column: series.column,
+        });
+      }
+
+      return series.seriesKey;
+    })
+    .filter(isNotNull);
+  const seriesColors = settings.series_settings
+    ? _.mapObject(settings.series_settings, value => {
+        return value.color;
+      })
+    : undefined;
+  const chartColors = getColorsForValues(keys, seriesColors, palette);
+
+  return multipleSeries.map((series, index) => {
+    return {
+      ...(_.omit(series, "column", "seriesKey") as Series),
```

Comment:
This is kind of optional, but would make sure `series` passed to the rest of the combo chart has the same exact shape as before (has no `column`, or `seriesKey` properties)